luarules/gadgets.lua

```diff
@@ -1319,13 +1399,18 @@ end
 
 function gadgetHandler:AllowCommand(unitID, unitDefID, unitTeam,
 									cmdID, cmdParams, cmdOptions, cmdTag, playerID, fromSynced, fromLua)
+	local cmdKey = cmdID and cmdID or CMD_NIL
```

Comment:
```suggestion
	local cmdKey = cmdID or CMD_NIL
```
But also I don't think cmdID should ever be nil.

Reply:
Probably not, but on the other hand some api user could be sending nil (for example if using a wrong CMD_XXX var, lua will silently send nil) and didn't want this to blow up. I did it mainly because I saw the gaia_critters.lua checking for that so I thought it must have happened at some point or maybe happening still.